mautic/mautic
# Added the ability to give permissions on the campaigns.

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Enhancement
| Automated tests included? | No
| Related user documentation PR URL | None
| Related developer documentation PR URL | None
| Issues addressed (#s or URLs) | [#6166](https://github.com/mautic/mautic/issues/6166)
| BC breaks? | No
| Deprecations? | No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR gives the alternative to set permissions in the roles concerning campaigns.


#### Steps to test this PR:
1. Apply the PR
2. As an administrator, create a role with only view own/edit own/create/delete own/publish own selected for the campaigns permissions and the necessary permissions on others categories in order to build a campaign.
3. Create 2 new users with this role.
4. Create a campaign as an administrator
5. Logout and login with one of the new user.
6. Go to the campaign tab and note that you don't see the previously created campaign (whose belongs to admin).
7. Create a campaign. 
8. Logout and login with the second user. Then, repeat the step 6. You don't be able to see neither the admin's campaign nor the first user's campaign.
9. Each action of delete/edit/clone are covered by the permissions so you can't delete/edit/clone the campaign of anyone else when you're logged in as one or the other users you've created, even if you try to change it by the url way.

NB: Testing this PR have permitted to find a bug concerning the clone of some objects when the "own" permissions were set. This bug is fixed in this PR [#6238](https://github.com/mautic/mautic/pull/6238). So the permissions enhancement is complete for the campaigns.
